cpp/ps0006/p00006.cpp

```diff
@@ -2,23 +2,84 @@
 using namespace std;
 
  void counter(int end,int start=0,int step=1){ 
-    if(step==0||(start>end && step>0)||(end>start && step<0)){
-        cout<<"Invalid input parameters. Please check the start, end, and step values." ;
-        return ;
+    if(step==0 && start==end){ // validating a test case
+        cout<<end<<endl;
+        return;
+    }
+    if(step==0||(start>end && step>0)||(end>start && step<0)||start==end||(end-start)%step!=0){ // validating some test cases
+        cout<<"Invalid input parameters. Please check the start, end, and step values.";
+        return;
     } 
-   if (step>0){ 
-        for( int i=start; i<=end;i+=step){
-            cout<< i <<endl ;
+    if (step>0){
+        for(int i=start;i<=end;i+=step){
+            cout<<i<<endl;
         }
     }
     else if(step<0){
         for(int i=start;i>=end;i+=step){
-            cout<< i<<endl;
+            cout<<i<<endl;
         }
-    }    
+    } // DRY!
  }
 
 int main()
 {
-   counter(10,0,1);
-}
\ No newline at end of file
+    // running my valid test cases 2/3
+    
+    counter(5); // 0 1 2 3 4 5
+    cout << "===============" << endl;
+    
+    counter(3, 1); // 1 2 3
+    cout << "===============" << endl;
+    
+    counter(-21, 6, -3); // 6 3 0 -3 -6 -9 -12 -15 -18 -21
+    cout << "===============" << endl;
+    
+    counter(10, 10, 0); // 10
+    cout << "===============" << endl;

```

Comment:
- [x] `start=0`, `step=1` as default
- [x] up counter works properly when `step` is positive
- [x] down counter works properly when `step` is negative
- [ ] `step=0` case, when the counter starts and ends at the same value, is handled